### PR TITLE
Use repos from state in catalog

### PR DESCRIPTION
**Pull request by antgamdia**
### Description of the change

This PR uses the `repos` property, which already exists in the global state, and adds the component state variable `currentRepo` for tracking the selected repos (used in the /chart request)  

### Benefits

The logic to getting all the repositories will no longer require iterate over the whole list of charts; instead, repos will be retrieved from the state.

### Possible drawbacks

N/A

### Applicable issues

This PR depends on https://github.com/kubeapps/kubeapps/pull/2249 to be merged.

### Additional information

- Change initially introduced in https://github.com/kubeapps/kubeapps/pull/2221 but removed from there to keep it as simple as possible.

### CODE REVIEW

**Comment by absoludity** [comment]
Did you forget to set the base branch on this PR, or is it really 24 commits? :) (Guessing you did, so will hold off).

**Comment by antgamdia** [comment]
> Did you forget to set the base branch on this PR, or is it really 24 commits? :) (Guessing you did, so will hold off).

Ups, sorry! This PR should be marked as a draft, I forgot to do so! (The few changes here are the ones from the commit https://github.com/kubeapps/kubeapps/pull/2251/commits/cd29e9ebb9e0db5c629695714ef88503de16ed96).
All my latest PRs just depends upon https://github.com/kubeapps/kubeapps/pull/2249

**Comment by antgamdia** [comment]
I have a failing test here, but before going deeper I just want to confirm you are ok with the approach, @andresmgot .

File: dashboard/src/components/Catalog/Catalog.tsx [hunk 1]
```diff
@@ -125,14 +135,31 @@ function Catalog(props: ICatalogProps) {
...
   ).sort();
 
+  // We do not currently support app repositories on additional clusters.
+  const supportedCluster = cluster === kubeappsCluster;
+  useEffect(() => {
+    if (!namespace) {
+      // All Namespaces
+      fetchRepos("");
+      return;
+    }
```
Comment (andresmgot) [inline dashboard/src/components/Catalog/Catalog.tsx:153]: not having a namespace is not a possibility in the Catalog view.

File: dashboard/src/components/Catalog/Catalog.tsx [hunk 2]
```diff
@@ -141,8 +168,8 @@ function Catalog(props: ICatalogProps) {
   // Only one search filter can be set
   const searchFilter = propsFilter[filterNames.SEARCH]?.toString() || "";
   useEffect(() => {
-    fetchCharts(cluster, namespace, repo, searchFilter);
-  }, [fetchCharts, cluster, namespace, repo, searchFilter]);
+    fetchCharts(cluster, namespace, currentRepo, searchFilter);
```
Comment (andresmgot) [inline dashboard/src/components/Catalog/Catalog.tsx:171]: why do you need to store a separate state variable? won't it work simply using the filters already in the state? Something like:

```js
const reposFilter = filters[filterNames.REPOS]?.join(",") || "";
 useEffect(() => {
    fetchCharts(cluster, namespace, reposFilter, searchFilter);
  }, [fetchCharts, cluster, namespace, reposFilter, searchFilter]);
```

**Review by andresmgot — COMMENTED** [review]
The approach makes sense to me but I think you don't need the new state var

File: dashboard/src/components/Catalog/Catalog.tsx (hunk 2 above)
Comment (antgamdia) [reply dashboard/src/components/Catalog/Catalog.tsx:171]: You're right, if we already have the `filters` in the state it doesn't really make sense. I wrongly thought that, perhaps, `reposFilter`  was not always referring to the current value,

File: dashboard/src/components/Catalog/Catalog.test.tsx [hunk 3]
```diff
@@ -212,15 +213,17 @@ describe("filters by application repository", () => {
...
+    const fetchRepos = jest.fn();
+    const wrapper = mountWrapper(store, <Catalog {...populatedProps} fetchRepos={fetchRepos} />);
     // The repo name is "foo"
     const input = wrapper.find("input").findWhere(i => i.prop("value") === "foo");
     input.simulate("change", { target: { value: "foo" } });
     // It should have pushed with the filter
     const historyAction = store
       .getActions()
       .find(action => action.type === "@@router/CALL_HISTORY_METHOD");
+    expect(fetchRepos).toHaveBeenCalledWith("kubeapps");
```
Comment (absoludity) [inline dashboard/src/components/Catalog/Catalog.test.tsx:226]: So this tests the case where the namespace is kubeapps, but we could also update a separate test to ensure that
```
expect(fetchRepos).toHaveBeenCalledWith("otherNamespace", true);
```
?

**Review by absoludity — APPROVED** [review]

File: dashboard/src/components/Catalog/Catalog.test.tsx (hunk 3 above)
Comment (andresmgot) [reply dashboard/src/components/Catalog/Catalog.test.tsx:226]: :heavy_plus_sign:

**Review by andresmgot — APPROVED** [review]
LGTM

**Pull request merged**